Gems/Atom/Asset/Shader/Code/CMakeLists.txt

```diff
@@ -49,8 +49,9 @@ ly_add_target(
     INCLUDE_DIRECTORIES
         PRIVATE
             Source
-            Source/Editor
             ${pal_source_dir}
+        PUBLIC
+            Source/Editor
```

Comment:
nit- ideally, only the feathers that need to get exposed are moved to Include/Shader/Editor

Reply:
Yes, I was actually undersatanding your suggestion under this angle. But then I looked at the family of source files as they are in this area of the code, and I was like "oh man, pulling this one sole header alone in its own folder is gonna look super odd and unjustifiably unjust to rip a kid off his siblings". ^^ I mean you get it, but I exposed the whole thing because I couldn't find a rationale for some feature exposition compared to the other. Maybe if that one asset struct was needed elsewhere, the other files share everything, they're undistinguishable in level of abstraction of "exposition rationale". So I just went with the big foot PUBLIC. The advantage is that it keeps file location cohesion and there is no need to explain "that particular file is needed elesewhere so it's in its own folder over there".